tests/tests_unit/graph/test_transformers/test_prune_graph.py

```diff
@@ -0,0 +1,141 @@
+from collections.abc import Iterable
+
+import pytest
+from _pytest.mark import ParameterSet
+from rdflib import Literal, Namespace, URIRef
+
+from cognite.neat._graph.transformers._prune_graph import TwoHopFlattener
+from cognite.neat._shared import Triple
+from cognite.neat._store import NeatGraphStore
+
+RDF_TYPE = URIRef("http://www.w3.org/1999/02/22-rdf-syntax-ns#type")
+
+
+def generate_test_parameters() -> Iterable[ParameterSet]:
+    value_property_name = "value"
+    predicate_property_name = "description"
+    namespace = Namespace("http://www.io-link.com/IODD/2010/10/")
+    destination_node_type = namespace["TextObject"]
+
+    triples_keep_old_predicate = [
+        (namespace["Text-Destination-ID"], RDF_TYPE, namespace["TextObject"]),
+        (namespace["Text-Destination-ID"], namespace["value"], Literal("Vacuum system self-check completed.")),
+        (namespace["Device-Source-ID"], namespace["textProperty"], namespace["Text-Destination-ID"]),
+        (namespace["Device-Source-ID"], RDF_TYPE, namespace["Device"]),
+    ]
+
+    expected_triples_keep_old_predicate = [
+        (namespace["Device-Source-ID"], RDF_TYPE, namespace["Device"]),
+        (namespace["Device-Source-ID"], namespace["textProperty"], Literal("Vacuum system self-check completed.")),
+    ]
+
+    expected_triples_keep_old_predicate = [
+        [str(item) for item in triple] for triple in expected_triples_keep_old_predicate
+    ]
+    expected_triples_keep_old_predicate.sort()
+
+    yield pytest.param(
+        triples_keep_old_predicate,
+        destination_node_type,
+        namespace,
+        value_property_name,
+        None,
+        expected_triples_keep_old_predicate,
+        id="Flatten and keep old predicates",
+    )
+
+    triples_new_predicate = [
+        (namespace["Text-Destination-ID"], RDF_TYPE, namespace["TextObject"]),
+        (namespace["Text-Destination-ID"], namespace["value"], Literal("Vacuum system self-check completed.")),
+        (namespace["Text-Destination-ID"], namespace["description"], Literal("vacuum status")),
+        (namespace["Device-Source-ID"], namespace["textProperty"], namespace["Text-Destination-ID"]),
+        (namespace["Device-Source-ID"], RDF_TYPE, namespace["Device"]),
+    ]
+
+    expected_triples_new_predicate = [
+        (namespace["Device-Source-ID"], RDF_TYPE, namespace["Device"]),
+        (namespace["Device-Source-ID"], namespace["vacuumStatus"], Literal("Vacuum system self-check completed.")),
+    ]
```

Comment:
I like this, makes it very nice to understand what is going to happen in the transformer.